lib/foreman_puppet_enc/engine.rb

```diff
@@ -15,19 +15,53 @@ class Engine < ::Rails::Engine
     initializer 'foreman_puppet_enc.register_plugin', before: :finisher_hook do |_app|
       Foreman::Plugin.register :foreman_puppet_enc do
         requires_foreman '>= 2.0.0'
-
         # Add Global JS file for extending foreman-core components and routes
         register_global_js_file 'fills'
 
         # Add permissions
-        # security_block :foreman_puppet_enc do
-        # end
+        security_block :foreman_puppet_enc do
+          permission :view_config_groups, { :config_groups => %i[index auto_complete_search welcome],
+                                            :"api/v2/config_groups" => %i[index show],
```

Comment:
this can be removed for now, but if you want to bring it before we have the actual API controllers, lets prefix it as well.
```suggestion
                                            :"foreman_puppet_enc/api/v2/config_groups" => %i[index show],
```